File: 0973-k-closest-points-to-origin/0973-k-closest-points-to-origin.py
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        heap = []

        for x, y in points:
            dist = (0 - x)**2 + (0 - y)**2
            heap.append((dist, x, y))

        heapq.heapify(heap)

        res = []

        while len(res) < k:
            res.append(list(heapq.heappop(heap)[1:]))

        return res

File: 0973-k-closest-points-to-origin/0973-k-closest-points-to-origin.go
func kClosest(points [][]int, k int) [][]int {
    var calcDist func(x int, y int) float64

    calcDist = func(x int, y int) float64 {
        xFloat := float64(x)
        yFloat := float64(y)
        return math.Sqrt(math.Pow(xFloat, 2) + math.Pow(yFloat, 2))
    }

    distances := [][]float64{}

    for idx, coords := range points {
        dist := calcDist(coords[0], coords[1])
        distances = append(distances, []float64{dist, float64(idx)})
    }

    sort.Slice(distances, func(i, j int) bool {
		return distances[i][0] < distances[j][0]
	})

    result := [][]int{}

    for i := 0; i < k; i++ {
        result = append(result, points[int(distances[i][1])])
    }

    return result
}